bin/resident

```diff
@@ -0,0 +1,202 @@
+    return False
+
+
+def get_params(arguments):
+    """Use given arguments to generate hook command.
+
+    Empty for now, options will be added.
+    """
+    return u"""{0} lint --file $1\n""".format(
+        os.path.abspath(inspect.getsourcefile(lambda: 0)))
+
+
+def upsert(hook_path, command, overwrite=False):
+    """Check if hook already exists.
+
+    Overwrites it or not depending on the setting.
+    """
+    if not os.path.isfile(hook_path):
+        return False
+
+    insterted = False
+    temp_fh, temp_path = mkstemp()
+    with open(temp_path, "w") as temp:
+        with io.open(hook_path) as old_file:
+            for line in old_file:
+                match = re.match(r"^\s*\S*(\S)resident ", line)
+                if (not insterted and match is not None and
+                                      match.group(1) == os.path.sep):
+                    if overwrite:
+                        temp.write(command)
+                    else:
+                        print("WARN:  resident is already installed in %s" %
+                              hook_path)
+                        temp.write(line)
+                    insterted = True
+                else:
+                    temp.write(line)
+    os.close(temp_fh)
+    if insterted:
+        os.remove(hook_path)
+        shutil.move(temp_path, hook_path)
+    else:
+        os.remove(temp_path)
+        with io.open(hook_path, "a") as hook_file:
+            hook_file.write(u"{0}{1}".format(command, os.linesep))
+    return True
+
+
+def create(hook_path, command):
+    """Create hook."""
+    with io.open(hook_path, "w") as hook:
+        hook.write(u"#!/usr/bin/env sh{0}{0}{1}{0}".format(
+            os.linesep, command))
+    return True
+
+
+def setup_hook(repo_path):
+    """Setup hoook in given git repository."""
+    hook_path = ensure_git_dir(repo_path)
+    if hook_path:
+        command = get_params(arguments)
+        (upsert(hook_path, command, arguments.force) or
+         create(hook_path, command))
+        st = os.stat(hook_path)
+        os.chmod(hook_path, st.st_mode | stat.S_IXUSR)
+    else:
+        raise ValueError(u"%s is not a git directory." % repo_path)
+
+
+def format_errrors(errors, file="your commit message."):
+    if len(errors) == 1:
+        print("The following error was found %s" % file)
+        print(errors[0])
+    else:
+        print("The following errors were found %s" % file)
+        print("\n".join(errors))
+
+
+def lint_files(file_paths):
+    errors = []
+    for file_path in file_paths:
+        with io.open(file_path) as message:
+            errors.append(rules.apply_rules(message.read()))
+    if (len(file_paths) == 1) and (len(errors[0]) != 0):
+        format_errrors(errors[0])
+        return False
+    else:
+        result = True
+        for file_path, file_errors in zip(file_paths, errors):
+            if len(file_errors) != 0:
+                format_errrors(file_errors, file_path)
+                result = False
```

Comment:
I thought i would check all the files and comment on all of them.
Avoids having to fix one file and then having the test fail on the following one?